packages/flutter_tools/lib/src/commands/analyze.dart

```diff
@@ -182,8 +182,16 @@ class AnalyzeCommand extends FlutterCommand {
           .where((String line) => !line.startsWith(new RegExp(r'^ *#')))
           .forEach((String line) {
             int colon = line.indexOf(':');
-            if (colon > 0)
-              dependencies.add(line.substring(0, colon), path.normalize(path.absolute(directory.path, path.fromUri(line.substring(colon+1)))), dotPackagesPath);
+            if (colon > 0) {
+              String packageName = line.substring(0, colon);
+              String packagePath = path.fromUri(line.substring(colon+1));
+              // Ensure that we only add the `analyzer` package defined in the vended SDK (and referred to with a local path directive).
+              // Analyzer package versions reached via transitive dependencies (e.g., via `test`) are ignored since they would produce
+              // spurious conflicts.
+              if (packageName != 'analyzer' || packagePath.startsWith('..')) {
```

Comment:
@pq  As of https://github.com/flutter/flutter/pull/5707, the `flutter analyze` command automatically detects if the files are in the flutter repo regardless of whether the `--flutter-repo` flag is specified. IOW `flutterRepo` bool == `--flutter-repo` flag specified on the command line -or- specified files (or current directory if no files specified) are in the flutter repo.